test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/BulkUpdateE2ETests.cs

```diff
@@ -0,0 +1,456 @@
+                expectedResponse);
+
+            var person = new Person
+            {
+                ID = 100,
+                Name = "Bing",
+            };
+
+            var person2 = new Person
+            {
+                ID = 200,
+                Name = "Edge"
+            };
+
+            var car1 = new Car { ID = 1001 };
+            var car2 = new Car { ID = 1002 };
+            var car3 = new Car { ID = 1003 };
+
+            this.context.AttachTo("Persons", person);
+            this.context.AttachTo("Persons", person2);
+            this.context.AttachTo("Cars", car1);
+            this.context.AttachTo("Cars", car2);
+            this.context.AttachTo("Cars", car3);
+
+            this.context.AddLink(person, "Cars", car1);
+            this.context.AddLink(person2, "Cars", car2);
+            this.context.AddLink(person2, "Cars", car3);
+
+            DataServiceResponse response = this.context.BulkUpdate(person, person2);
+
+            Assert.Equal(2, response.Count());
+
+            var personOperationResponse = response.FirstOrDefault() as ChangeOperationResponse;
+            var person2OperationResponse = response.LastOrDefault() as ChangeOperationResponse;
+
+            Assert.Single(personOperationResponse.NestedResponse);
+            Assert.Equal(2, person2OperationResponse.NestedResponse.Count);
+        }
+
+        [Fact]
+        public void DescriptorForAFailedCreateOperation_NotUpdated()
+        {
+            var expectedResponse = "{\"@context\":\"http://localhost:8000/$metadata#Persons/$delta\",\"value\":[{\"@removed\": {\"reason\": \"changed\"},\"@id\": \"http://localhost:8000/Persons(100)\",\"@Core.DataModificationException\": {\"@type\": \"#Org.OData.Core.V1.DataModificationExceptionType\"},\"Id\": 100,\"Name\": \"Bing\"}]}";
+            
+            SetupContextWithRequestPipelineForSaving(
+                this.context,
+                expectedResponse);
+
+            var person = new Person
+            {
+                ID = 100,
+                Name = "Bing",
+            };
+
+            this.context.AttachTo("Persons", person);
+
+            DataServiceResponse response = this.context.BulkUpdate(person);
+
+            var personChangeOperationResponse = response.FirstOrDefault() as ChangeOperationResponse;
+
+            Assert.Equal(EntityStates.Unchanged, personChangeOperationResponse.Descriptor.SaveResultWasProcessed);
+        }
+
+        [Fact]
+        public void DescriptorsWithNullEntitySetName_ShouldBeSerializedSuccessfully()
+        {
+            SetupContextWithRequestPipelineForSaving(
+                this.context,
+                persons);
+
+            var person = this.context.Persons.Execute().FirstOrDefault();
+
+            var personEntityDescriptor = this.context.GetEntityDescriptor(person);
+
+            Assert.Null(personEntityDescriptor.EntitySetName);
+
+            this.bulkUpdateGraph = this.GetBulkUpdateGraph(person);

```

Comment:
If you're modifying this instance field in a test method, does it make sense to have it as an instance field? Could it just have been a local variable?